backend/lib-shared-site-api/src/db/util.rs

```diff
@@ -129,6 +135,63 @@ where
     (query, count)
 }
 
+pub fn append_column_info_to_query<'a, DB: Database>(
+    mut query: QueryBuilder<'a, DB>,
+    action_type: Action,
+    columns: &HashMap<String, ColumnInfo>,
+) -> QueryBuilder<'a, DB> {
+    for (column_name, column_info) in columns.iter() {
+        let column_type = column_info.data_type.to_string();
+
+        match action_type {
+            Action::CreateTable => {
+                query.push(format!(", {}", column_name));
+            }
+            Action::AddColumn => {
+                query.push(column_name);
+            }
+            _ => {}
+        }
+
+        query.push(" ");
+        query.push(column_type);
+        query.push(" NOT NULL");
+
+        match action_type {
+            Action::AddColumn => {
+                query.push(" DEFAULT ''");
+            }
+            _ => {}
+        }
+
+        for rule in column_info.validation_rules.iter() {
+            match rule.rule_type {
+                RuleType::Unique => {
+                    query.push(" UNIQUE");
+                }
+                RuleType::MinLength => {
```

Comment:
When testing, I encountered an issue. If I add a column with a `MinLength` constraint, but because our default value is empty, this will cause a conflict. Perhaps, as you mentioned earlier, it would be better to execute validation conditions from the frontend? If we don't need the database to handle this, we can remove the code here.

Reply:
Thanks! Yes, I will consider moving validation to the application code. It seems convenient to have `UNIQUE` in the DB, but probably simpler overall to keep everything in the application.